ocaml/xenopsd/lib/xenops_server.ml

```diff
@@ -3777,8 +3777,14 @@ let upgrade_internal_state_of_running_vms () =
   let module B = (val get_backend () : S) in
   List.iter
     (fun vm ->
-      let vm_t = VM_DB.read_exn vm in
-      B.VM.get_internal_state [] [] vm_t |> B.VM.set_internal_state vm_t
+      try
+        let vm_t = VM_DB.read_exn vm in
+        B.VM.get_internal_state [] [] vm_t |> B.VM.set_internal_state vm_t
+      with
+      | Xenops_interface.Xenopsd_error (Does_not_exist _) ->
+          warn "Missing VM extra metadata for VM %s" vm
+      | _ ->
+          warn "VM metadata upgrade failed for VM %s" vm
```

Comment:
might be worth printing the error here, are there exception serializers in Xenops_interface?

Reply:
Yes, and registered with `Printexc.register_printer`, so I think that gets picked up by `Printexc.to_string`.